src/main/java/com/spotify/futures/CompletableFutures.java

```diff
@@ -237,4 +242,41 @@ private CompletableFutures() {
         .thenApply(ignored ->
                        function.apply(af.join(), bf.join(), cf.join(), df.join(), ef.join()));
   }
+
+  /**
+   * Periodically poll an external resource until it returns a non-empty result.
+   *
+   * The polling task should return Optional.empty() until it becomes available, and then
+   * Optional.of(result).
+   *
+   * If the polling task throws an exception or returns null, that will cause the result future to
+   * complete exceptionally.
+   *
+   * @param pollingTask the polling task.
+   * @param period the polling period.
+   * @param unit the polling time unit.
+   * @param executorService the executor service to schedule the polling task on.
+   * @return a future completing to the result of the polling task once that becomes available.
+   */
+  public static <T> CompletableFuture<T> poll(
+      final Supplier<Optional<T>> pollingTask,
+      final long period, final TimeUnit unit,
```

Comment:
I know that this is what people used to do and is commonly understood, but since we're already in java 8 land here, could we instead pass in one single `java.time.Duration frequency` parameter instead? I personally like it in APIs. Then the actual call to `scheduleAtFixedRate` call could take `frequency.toMillis(), TimeUnit.MILLISECONDS` for example.